src/Authenticator/LoginHandler.php

```diff
@@ -155,9 +209,52 @@ public function startRegister(HTTPRequest $request)
      * @param HTTPRequest $request
      * @return HTTPResponse
      */
-    public function finishRegister(HTTPRequest $request)
+    public function finishRegistration(HTTPRequest $request)
     {
-        return $this->jsonResponse(['errors' => ['Registration not yet implemented']], 500);
+        $sessionStore = $this->getSessionStore();
+        $sessionMember = $sessionStore->getMember();
+        $loggedInMember = Security::getCurrentUser();
+
+        if (is_null($loggedInMember) && is_null($sessionMember)) {
+            return $this->jsonResponse(
+                ['errors' => [_t(__CLASS__ . '.NOT_AUTHENTICATING', 'You must be logged or logging in')]],
+                403
+            );
+        }
+
+        $storedMethodName = $sessionStore->getMethod();
+        $specifiedMethodName = str_replace('-', '\\', $request->param('Method'));
+
+        // If a registration process hasn't been initiated in a previous request, calling this method is invalid
+        if (!$storedMethodName) {
+            return $this->jsonResponse(
+                ['errors' => [_t(__CLASS__ . '.NO_REGISTRATION_IN_PROGRESS', 'No registration in progress')]],
+                400
+            );
+        }
+
+        if ($storedMethodName !== $specifiedMethodName) {
+            return $this->jsonResponse(
+                ['errors' => [_t(__CLASS__ . '.METHOD_MISMATCH', 'Method does not match registration in progress')]],
+                400
+            );
+        }
+
+        /** @var MethodInterface $method */
+        $method = Injector::inst()->get($storedMethodName);
+        $registrationHandler = $method->getRegisterHandler();
```

Comment:
Similarly to the start handler, maybe you could to a type check here?

Reply:
No longer required, as `$method` will always be a valid object now.